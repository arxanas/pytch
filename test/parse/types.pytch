val foo :: int
type foo = int
type bar = (int, int)
type baz = {
  qux :: int
}
type grault = {
  xyzzy :: int,
  plugh :: int,
}
type corge = (int -> (int, int) -> bool) -> bool
