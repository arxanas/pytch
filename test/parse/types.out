(Program
  (ValStmt
    (Ident 'foo')
    (TypeExprAtom
      (Ident 'int')))
  (TypeStmt
    (Ident 'foo')
    (TypeExprAtom
      (Ident 'int')))
  (TypeStmt
    (Ident 'bar')
    (TupleTypeExpr
      (TypeExprAtom
        (Ident 'int'))
      (TypeExprAtom
        (Ident 'int'))))
  (TypeStmt
    (Ident 'baz')
    (RecordTypeExpr
      (RecordField
        (Ident 'qux')
        (TypeExprAtom
          (Ident 'int')))))
  (TypeStmt
    (Ident 'grault')
    (RecordTypeExpr
      (RecordField
        (Ident 'xyzzy')
        (TypeExprAtom
          (Ident 'int')))
      (RecordField
        (Ident 'plugh')
        (TypeExprAtom
          (Ident 'int'))))))
